pkg/tsdb/cloudwatch/cloudwatch.go

```diff
@@ -74,16 +76,97 @@ type cloudWatchExecutor struct {
 
 func (e *cloudWatchExecutor) newSession(region string) (*session.Session, error) {
 	dsInfo := e.getDSInfo(region)
-	creds, err := getCredentials(dsInfo)
+
+	cacheKey := fmt.Sprintf("%d:%s:%s:%s:%s", dsInfo.AuthType, dsInfo.AccessKey, dsInfo.Profile, dsInfo.AssumeRoleARN,
+		dsInfo.ExternalID)
+
+	sessCacheLock.RLock()
+	if env, ok := sessCache[cacheKey]; ok {
+		if env.expiration.After(time.Now().UTC()) {
+			sessCacheLock.RUnlock()
```

Comment:
I haven't followed a lot the discussion about the logging but I would like to get a debug message if a cached session is used.

Reply:
@papagian Agree, committed and pushed.